### DISCUSSION TITLE
What does DGGML_NATIVE build option do?

### DISCUSSION THREAD

**Discussion by abitrolly** [ggml-org/llama.cpp#10230] [Q&A] [answered]
I see `-DGGML_NATIVE=OFF` in several build scripts, but can not find the description in expected places like https://github.com/ggerganov/llama.cpp/blob/master/docs/build.md

What does it do?

**Comment by danbev** [comment, marked as answer]
This [disables](https://github.com/ggerganov/llama.cpp/blob/master/ggml/CMakeLists.txt#L70) `-march=native` which is a GCC/Clang flag used to specify that the compiler should optimize for the host processor.  If this set to true it enables all instruction set extensions supported by the host processor. The compiler probes the host system's processor for its capabilities like using `cpuid` and then adds then appropriate compiler flags that are available on the current system. 

Portability will be reduced as if the target machine does not support the same instructions, running the binary will lead to an illegal instruction error.
